Documentation/ceph-dashboard-monitoring.md

```diff
@@ -0,0 +1,329 @@
+3. [Grafana](#grafana): it will allow us to represent graphically the information collected by Prometheus
+4. [Prometheus Node exporter](#prometheus-node-exporter): provides cluster nodes machine metrics to Prometheus
+5. [Prometheus alert manager](#prometheus-alert-manager): manage alerts produced in Prometheus
+
+
+## Prerequisites
+- A working Ceph cluster deployed using the Rook operator.
+- [Prometheus manager module must be enabled](https://docs.ceph.com/en/latest/mgr/prometheus/#enabling-prometheus-output)
+- [Dashboard must be enabled](ceph-dashboard.md#enable-the-ceph-dashboard) in the Ceph cluster CRD
+- A PV with available space in order to store Grafana database (500Mb)
+- The files with the different resources definition and configuration. You can get them from the Rook repository:
+```console
+git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
+cd rook/cluster/examples/kubernetes/ceph/monitoring
+```
+## Prometheus Operator
+Follow the instructions provided in [Prometheus Monitoring](https://github.com/rook/rook/blob/{{ branchName }}/Documentation/ceph-monitoring.md)
+
+The Prometheus operator allows you to set up a Prometheus instance in you kubernetes cluster, a summary of the commands you will need to execute is below:
+
+Install the Prometheus Operator
+```console
+kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/v0.40.0/bundle.yaml
+```
+Go to the folder where the different resource files for the monitoring stack are stored
+```console
+cd rook/cluster/examples/kubernetes/ceph/monitoring
+```
+Create and configure the Prometheus instance 
+```console
+kubectl create -f service-monitor.yaml
+kubectl create -f prometheus.yaml
+kubectl create -f prometheus-service.yaml
+```
+
+With these steps you must have a running instance of Prometheus configured to scrape data from the Ceph cluster through the Ceph Prometheus manager module
+
+To verify that prometheus is running you can try to access the prometheus instance.
+
+Execute:
+```console
+echo "http://$(kubectl -n rook-ceph -o jsonpath={.status.hostIP} get pod prometheus-rook-prometheus-0):30900"
+```
+>```
+>http://192.168.122.70:30900
+>```
+If you use the output URL in your browser, you should see the Prometheus monitoring website.
+
+![Prometheus Monitoring Website](media/prometheus-monitor.png)
+
+## Prometheus Ceph manager module
+The source of the Ceph cluster metrics is this manager module, by default is enabled, and the Prometheus operator service monitor configures Prometheus to scrape data from it.
+
+In order to avoid monitoring issues just verify that the scrape intervals are the same in Prometheus and in the Ceph manager module. Both intervals must be the same, depending on your monitoring needs, a value between 5 and 30 is correct.
+
+In the Prometheus web UI, go to Status-Configuration, the scrape configuration should be like:
+
+```yaml
+scrape_configs:
```

Comment:
Can we just see the value with the `ceph config get` command below? If so, seems like we don't need to document this step.

Reply:
The "scrape configs" are part of the Prometheus configuration. Is not possible to get this information using a "ceph config get" command